src/test/ui/async-await/issue-75785-confusing-named-region.rs

```diff
@@ -0,0 +1,13 @@
+// edition:2018
+//
+// Regression test for issue #75785
+// Tests that we don't point to a confusing named
+// region when emitting a diagnostic
+
+pub fn async_fn(x: &mut i32) -> (&i32, &i32) {
```

Comment:
This needs to be `async` to repro the issue, doesn't it? The output for this is unchanged from the current output.

Reply:
Oops, thanks!